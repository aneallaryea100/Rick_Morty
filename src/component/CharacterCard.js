import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function CharacterCard(props) {
  const {
    id, image, name, handleDetails,
  } = props;
  return (
    <div key={id}>
      <p>{name}</p>
      <img src={image} alt={name} />
      <button type="button" onClick={handleDetails}>
        <Link className="detailedLink" id={id} to={`/details/${id}`}>Details</Link>
      </button>
    </div>
  );
}

CharacterCard.propTypes = {
  id: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleDetails: PropTypes.func.isRequired,
};

// handleDetails
export default CharacterCard;
