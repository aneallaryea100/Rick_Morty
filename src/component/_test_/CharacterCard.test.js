import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CharacterCard from '../CharacterCard';
import store from '../../redux/configureStore';

describe('CharacterCard test', () => {
  const setCharId = store.state;
  const handleDetails = (event) => {
    const btn = event.target.id;
    setCharId(Number(btn));
  };
  test('snapshot', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CharacterCard handleDetails={handleDetails} />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
