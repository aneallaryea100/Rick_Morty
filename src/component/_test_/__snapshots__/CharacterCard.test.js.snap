// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard test snapshot 1`] = `
<div
  className="cardCharacters1"
>
  <div
    className="imageContainer"
  >
    <img
      className="imageCharacterCard"
    />
  </div>
  <div
    className="cardFooter"
  >
    <p
      className="nameCharacter"
    />
    <button
      className="clickBtnCard"
      onClick={[Function]}
      type="button"
    >
      <a
        className="detailedLink"
        href="/details/undefined"
        onClick={[Function]}
      >
        Details
      </a>
    </button>
  </div>
</div>
`;
